rdt-app/src/main/java/io/ona/rdt_app/interactor/PatientRegisterFragmentInteractor.java

```diff
@@ -167,4 +176,40 @@ public Patient getPatientForRDT(JSONObject jsonFormObject) throws JSONException
         return rdtPatient;
     }
 
+    public synchronized void launchForm(Activity activity, String formName, Patient patient) throws JSONException {
+        try {
+            JSONObject formJsonObject = formUtils.getFormJsonObject(formName, activity);
+            if (patient != null) {
+                FormLaunchArgs args = new FormLaunchArgs().withActivity(activity)
+                        .withPatient(patient)
+                        .withFormJsonObj(formJsonObject);
+                formUtils.getNextUniqueId(args, this);
```

Comment:
Why get a uniqueId if the patient exists

Reply:
The unique id is for the rdt not patient.

You should only need to generate this id when launching the rdt test form, which would have a patient associated with it.